pkg/ipam/node_manager.go

```diff
@@ -418,15 +418,14 @@ func (n *NodeManager) Get(nodeName string) *Node {
 // allocated or released, sorted by the number of addresses needed to be operated
 // in descending order. Number of addresses to be released is negative value
 // so that nodes with IP deficit are resolved first
+// The caller must hold the NodeManager lock
 func (n *NodeManager) GetNodesByIPWatermark() []*Node {
```

Comment:
In such cases, the convention is to call the function xxxLocked(...) instead of just having a comment.

ex. https://github.com/tommyp1ckles/cilium/blob/main/pkg/aws/eni/node.go#L92